pandas/tests/test_indexing.py

```diff
@@ -5619,19 +5650,12 @@ def f():
                           slice(3, 4.0),
                           slice(3.0, 4.0)]:
 
-                    def f():
-                        s.iloc[l]
-                    self.assertRaises(TypeError, f)
-
-                    def f():
-                        s.loc[l]
-                    self.assertRaises(TypeError, f)
+                    for idxr in [lambda x: x.loc,
+                                 lambda x: x.iloc,
+                                 lambda x: x]:
+                        self.assertRaises(TypeError, lambda: idxr(s)[l])
```

Comment:
Yes, but why would slicing be an exception?

```
In [107]: s.loc[1.0]
Out[107]: 1

In [108]: s.loc[2.0]
Out[108]: 2

In [109]: s.loc[1.0:2.0]
TypeError: cannot do slice indexing on <class 'pandas.indexes.numeric.Int64Index
'> with these indexers [1.0] of <type 'float'>
```

In ``s[1.0:2]``, the `1.0` is an integer location, so that should indeed raise, but in `s.loc[1.0:2]`, the `1.0` is a label.